lib/r10k/forge/module_release.rb

```diff
@@ -108,12 +115,70 @@ def download
       def verify
         logger.debug1 "Verifying that #{@tarball_cache_path} matches checksum"
 
-        md5_of_tarball = Digest::MD5.hexdigest(File.read(@tarball_cache_path, mode: 'rb'))
+        sha256_of_tarball = Digest::SHA256.file(@tarball_cache_path).hexdigest
+
+        if @sha256_file_path.exist?
+          verify_from_sha256_file(sha256_of_tarball)
+        else
+          begin
+            verify_sha256_from_forge(sha256_of_tarball)
+          rescue R10K::Error
```

Comment:
This is just using an exception for control flow and should be done differently.